subprojects/core/src/test/groovy/org/gradle/internal/work/DefaultAsyncWorkTrackerTest.groovy

```diff
@@ -119,8 +117,8 @@ class DefaultAsyncWorkTrackerTest extends ConcurrentSpec {
                 asyncWorkTracker.waitForCompletion(operation1, RELEASE_PROJECT_LOCKS)
                 instant.waitFinished
             }
+            thread.blockUntil.waitStarted
```

Comment:
I can see how this fixes the test, though I am not sure if the test is now testing what it should be testing. It seems like it still does, though I am not 100% sure.

Reply:
Previously the `workerThread` logic was simply done in a thread and not a worker. With the workers limited to one, ordering matters. That is if the blocking one is scheduled first, it prevents the other ones to run.
Moving the blocking operation _before_ preserves the test logic.